spring-integration-mail/src/main/java/org/springframework/integration/mail/AbstractMailReceiver.java

```diff
@@ -553,9 +595,25 @@ Store getStore() {
 
 		private final MimeMessage source;
 
+		private final Object content;
+
 		private IntegrationMimeMessage(MimeMessage source) throws MessagingException {
 			super(source);
 			this.source = source;
+			if (AbstractMailReceiver.this.simpleContent) {
+				this.content = null;
+			}
+			else {
+				Object complexContent;
+				try {
+					complexContent = source.getContent();
+				}
+				catch (IOException e) {
+					complexContent = "Unable to extract content; see logs: " + e.getMessage();
+					AbstractMailReceiver.this.logger.error("Failed to extract content from " + source, e);
+				}
+				this.content = complexContent;
```

Comment:
Because the folder needs to be open to retrieve the content - that's the whole reason for this class to exist.